src/audio_pixel.cpp

```diff
@@ -0,0 +1,270 @@
+#include "audio_pixel.h"
+#define project nullptr
+
+// audio pixel functions 
+
+double linear_interp(double x, double x1, double x2, double y1, double y2) {
+    // edge cases 
+    if ((x2-x1) == 0)
+        return x1;
+    // interpolation in time domain
+    return (((x2 - x) / (x2 - x1)) * y1) + ((x - x1) / (x2 - x1)) * y2;
+}
+
+double closest_val(double val1, double val2, double target) {
+    return abs(target - val1) <= abs(target - val2) ? val1 : val2;
+}
+
+// ************* audio_pixel_block_t ****************
+
+bool audio_pixel_block_t::flush() const {
+    json j;
+    to_json(j);
+    std::string resource_path = GetResourcePath();
+    std::ofstream ofs;
+    ofs.open(resource_path + "/kiwi-block.json");
```

Comment:
this works on windows? like the filepath with a forward slash?

Reply:
yep! i haven't had an issues with the slashes

Reply:
i tried it across my desktop and laptop